### DISCUSSION TITLE
End of Life for Windows PowerShell 5.1

### DISCUSSION THREAD

**Discussion by pregazzi** [PowerShell/PowerShell#16971] [General]
Will there be an end-of-life or an end-of-support for Windows PowerShell 5.1?
According to this article https://docs.microsoft.com/en-us/powershell/scripting/install/powershell-support-lifecycle?view=powershell-7.2 there does not seem to be any.
Can anyone advise?
Thanks

**Comment by kilasuit** [comment]
This note is at the top of the article you mentioned that states about how it it supported as a part of each OS it is a component of, so it's EOL/EOS is what ever that EOL/EOS is for the underlying OS that it is running on.. 
![image](https://user-images.githubusercontent.com/6355225/157116567-bb28674a-ac9d-4b10-a917-208dcd0c4bd1.png)

**Comment by pregazzi** [comment]
Thanks Ryan,

So, just for clarity, which OS will cause PS 5.1 to reach its EOL/EOS?

Both Windows 10 and 11 are supplied with WindowsPowershell 5.1 and both allow Powershell 7 too.

On a more generic level, when will PS 5.1 cease to be available and all scripts can only run on PS 7?



On 7 Mar 2022, at 21:03, Ryan Yates ***@***.******@***.***>> wrote:


CAUTION: This is an EXTERNAL email and has originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.




This note is at the top of the article you mentioned that states about how it it supported as a part of each OS it is a component of, so it's EOL/EOS is what ever that EOL/EOS is for the underlying OS that it is running on..
[image]<https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F6355225%2F157116567-bb28674a-ac9d-4b10-a917-208dcd0c4bd1.png&data=04%7C01%7Cpaolo.regazzi%40truata.com%7C1744a558c43744cf4b5c08da007def02%7C40f87ab9f24c470d829464306a0ea004%7C0%7C0%7C637822838139687477%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=RGYsvNVTRa3HwVUJPhLLOAG7wG9Ku40GL07%2BR%2B%2BOpbs%3D&reserved=0>

—
Reply to this email directly, view it on GitHub<https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FPowerShell%2FPowerShell%2Fdiscussions%2F16971%23discussioncomment-2311755&data=04%7C01%7Cpaolo.regazzi%40truata.com%7C1744a558c43744cf4b5c08da007def02%7C40f87ab9f24c470d829464306a0ea004%7C0%7C0%7C637822838139687477%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=rtWYFeKcXaMruqYGV78XmdPBghFwGw9i8PSEdHBWtPk%3D&reserved=0>, or unsubscribe<https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FARRNEMBMOLQXTOAROGVS3G3U6ZVKDANCNFSM5QEL4BCA&data=04%7C01%7Cpaolo.regazzi%40truata.com%7C1744a558c43744cf4b5c08da007def02%7C40f87ab9f24c470d829464306a0ea004%7C0%7C0%7C637822838139843710%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=JEl40VgmbW4SI2KGcD9PtarW5RUXGCZbC5ul0sDgwnI%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cpaolo.regazzi%40truata.com%7C1744a558c43744cf4b5c08da007def02%7C40f87ab9f24c470d829464306a0ea004%7C0%7C0%7C637822838139843710%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=T7dgpAegMrcHf2RnKdSZ5Sg3CB9735GiaF3O39QWFgY%3D&reserved=0> or Android<https://eur04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cpaolo.regazzi%40truata.com%7C1744a558c43744cf4b5c08da007def02%7C40f87ab9f24c470d829464306a0ea004%7C0%7C0%7C637822838139843710%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=hqKqBRy1p9W7I4WfveMgzoE5QfuzrF4PSUYhRn9cCwU%3D&reserved=0>.

**Comment by ADMDTO-Jack** [comment]
Very late answer, but based on the first response, since windows powershell 5.1 is included as a component of win11, it is supported for at least until the end of life of win11

https://learn.microsoft.com/en-us/lifecycle/products/windows-11-enterprise-and-education
![image](https://github.com/PowerShell/PowerShell/assets/173275703/c9fa48dd-e6ca-48bf-b194-c485557de537)

per the above image, windows 11 23H2 is supported until at least Nov 10, 2026, but also follows the "modern lifecycle policy", which means microsoft will give at minimum a 1 year notice before ending support (of win11 23h2 which bundles in powershell 5.1) if there is no next version (24H2, probably).
https://learn.microsoft.com/en-us/lifecycle/policies/modern
![image](https://github.com/PowerShell/PowerShell/assets/173275703/0812daa3-ee2f-4b90-aeff-70458b3640d9)

For practical purposes, 24H2 is probably going to come out this year, so that detail is unlikely to be relevant. Therefore, for 23H2, end of support is Nov 10, 2026; and if 24H2 contains powershell 5.1 as a component, then the earliest possible end of support for 5.1 is likely also pushed out another year. But if 24H2 does not contain powershell 5.1, then that Nov 2026 date could become a possibility.

Given that powershell 5.1 has a lot of value in legacy scripting, and powershell 7+ has imperfect backwards compatibility, I think it is likely that 5.1 continues to be bundled, or if not bundled, continues to be supported for an extended timeframe. My prediction is that if Microsoft did end 5.1 support as early as it is currently possible to do so, the community response would pressure them into formalizing an extended support timeframe of many more years.

But it is technically possible that the next version of windows (11,12,etc) does not come with 5.1, and on that day the expiration deadline will stop being pushed out automatically. At that time they will probably migrate to a formal EOL deadline for 5.1. Until that time, 5.1 can still be presumed to be supported indefinitely

(but I could be wrong, just my read on the situation. It's probably impossible to get a truly clear answer on this until Microsoft publishes a formal answer. Also apologies on the very late response, but I found this post while looking for an answer myself, so hopefully this helps the next person who asks the question too)

**Reply by pregazzi** [reply]
Thank you for the detailed answer, really appreciated.

**Comment by TiloGit** [comment]
maybe easier to track against Win server. 
Win 2022 server has EOL of **Oct 14, 2031** and ships with PS 5.1.
So I assume PS 5.1 will be supported until then or later (didn't check Win Server 2025 default PS yet)
